SnappyAPITests/PhotoViewControllerTests.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  PhotoViewControllerTests.swift
+//  Snappy
+//
+//  Created by Pawel Chmiel on 18.10.2016.
+//  Copyright © 2016 Droids On Roids. All rights reserved.
+//
+
+import XCTest
+@testable import Snappy
+
+class PhotoViewControllerTests: XCTestCase {
+   
+    var viewController  : PhotoViewController?
+   
+    override func setUp() {
+        super.setUp()
+        if viewController == nil {
+            let storyboard = UIStoryboard(name: "Main", bundle: Bundle.main)
+            viewController = storyboard.instantiateViewController(withIdentifier: "middle") as? PhotoViewController
+            UIApplication.shared.keyWindow?.rootViewController = viewController
+            
+            let _ = viewController?.view
+        }
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+    }
+    
+    override func tearDown() {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        super.tearDown()
+    }
+    
+    func testRemoveCameraPreview() {
+        if let firstSubview = viewController?.view.subviews.first {
+            viewController?.removeCameraPreview()
+            XCTAssertFalse((viewController?.view.subviews.contains(firstSubview))!)
+        }
+    }
+    
+    func testInsertCameraPreview() {
+        let someView = UIView()
+        viewController?.insertCameraPreview(someView)
+        XCTAssertTrue((viewController?.view.subviews.contains(someView))!)
+    }
+    
+    func testZHidePhoto() {
+        if let viewController = viewController, let lastSubview = viewController.view.subviews.last {
+            viewController.hidePhoto()
+            RunLoop.current.run(until: Date().addingTimeInterval(2.0))
+            XCUIDevice.shared().press(.home)
+
+            XCTAssertFalse((viewController.view.subviews.contains(lastSubview)))
```

Comment:
double parenthesis probably not needed here, please check everywhere where it is needed and where it is not